frontend/controller/actions/group.js

```diff
@@ -897,6 +897,61 @@ export default (sbp('sbp/selectors/register', {
       sbp('okTurtles.events/emit', OPEN_MODAL, 'AddMembers')
     }
   },
+  'gi.actions/group/removeUselessIdentityContracts': function ({ contractID, possiblyUselessContractIDs }) {
+    // NOTE: should remove the identity contracts which we don't need to sync anymore
+    //       for users who don't have any common groups, and any common DMs
+    const interval = setInterval(() => {
+      const pending = Object.entries(sbp('okTurtles.eventQueue/queuedInvocations'))
+        .filter(([q]) => typeof q === 'string')
```

Comment:
What exactly does this line do and why is it here?

I think this would work without it... and possibly be safer too.

Reply:
The `pending` variable would be filled with pending invocations. Since we should access the state of other contracts when we are leaving the group, we should make it clear that those contracts are all up to date. So just wait until this `pending` should be an empty list.

The reason why we could not use `await sbp('chelonia/contract/wait', contractIDs)` is that this has a potential deadlock. I need another solution that doesn't use `setInterval` (I knew I shouldn't use it, but could not find another solution), and I will find the better solution with the help of @corrideat.

Reply:
Sorry, I wasn't very clear, I meant why is this checking for `typeof q === 'strings'`? I believe they should all be strings.